### PR TITLE
DIG/SUGAR CreatedByAware

### PR DESCRIPTION
**Pull request by nick-prat**
Adding this to make them SPID Aware

### CODE REVIEW

File: src/foam/nanos/dig/DIGOwnerRuleAction.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
```
Comment (jlhughes) [inline src/foam/nanos/dig/DIGOwnerRuleAction.js]: Is this needed, can we not use 'createdBy' as that has both createdBy and createdByAgent.

**Review by jlhughes — COMMENTED** [review]
See inline comment regarding using createdBy and CreatedByAgent. 
what is the purpose of the 'owner' property?

File: src/services [hunk 2]
```diff
@@ -516,7 +516,33 @@ p({
...
+  "class":"foam.nanos.boot.NSpec",
+  "name":"digDAO",
+  "serve":true,
+  "serviceScript":
+  """
+  return new foam.dao.EasyDAO.Builder(x)
+    .setPm(true)
+    .setGuid(true)
+    .setOf(foam.nanos.dig.DIG.getOwnClassInfo())
+    .setCreatedByAware(true)
```
Comment (jlhughes) [inline src/services]: This is not necessary.

File: src/foam/nanos/dig/DIG.js [hunk 3]
```diff
@@ -17,6 +17,7 @@ foam.CLASS({
 
   implements: [
     'foam.mlang.Expressions',
+    'foam.nanos.auth.CreatedByAware'
```
Comment (jlhughes) [inline src/foam/nanos/dig/DIG.js]: Didn't realize we didn't have createdBy, created, lastModified, ... 
Please add the other three.

File: src/foam/nanos/dig/DIG.js [hunk 4]
```diff
@@ -40,10 +44,66 @@ foam.CLASS({
...
       name: 'id',
-      label: 'Request Name'
+      label: 'Request Name',
+      section: 'details'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'createdBy',
+      storageOptional: true,
```
Comment (nanoscott) [inline src/foam/nanos/dig/DIG.js]: created* properties shouldn't be storageOptional, only lastModified* properties.

**Review by nanoscott — CHANGES_REQUESTED** [review]

**Review by jlhughes — APPROVED** [review]

**Review by nanoscott — APPROVED** [review]

**Pull request merged**